packages/mui-styles/src/withStyles/withStyles.d.ts

```diff
@@ -37,10 +37,12 @@ export type BaseCreateCSSProperties<Props extends object = {}> = {
 
 export interface CreateCSSProperties<Props extends object = {}>
   extends BaseCreateCSSProperties<Props> {
+  variants?: Array<{ props: Props; style: CSSProperties }>;
   // Allow pseudo selectors and media queries
   [k: string]:
     | BaseCreateCSSProperties<Props>[keyof BaseCreateCSSProperties<Props>]
-    | CreateCSSProperties<Props>;
+    | CreateCSSProperties<Props>
+    | Array<{ props: Props; style: CSSProperties }>;
```

Comment:
Why is this added here? We don't be allowing variants to be deeply nested.

Reply:
Let me revert this, it is done to unrelated package.

Reply:
I moved the files to the correct place.